request_provider/__brick__/request_notifier.dart

```diff
@@ -6,17 +6,17 @@ import 'package:loggy/loggy.dart';
 
 import 'request_state.dart';
 
-abstract class RequestProvider<Value> extends StateNotifier<RequestState<Value, Exception>> with NetworkLoggy {
-  RequestProvider({RequestState<Value, Exception> initial = const RequestState.initial()}) : super(initial);
+abstract class RequestNotifier<Value> extends StateNotifier<RequestState<Value>> with NetworkLoggy {
+  RequestNotifier({RequestState<Value> initial = const RequestState.initial()}) : super(initial);
 
   Future<void> executeRequest({
     required ValueGetter<Future<Value>> requestBuilder,
-    Exception? Function(Exception)? errorHandler,
+    Exception? Function(Object)? errorHandler,
```

Comment:
The signature of `errorHandler` is a bit confusing (take a look at the underlying implementation of `executeRequest`).

1. The object passed to `errorHandler` will always be an `Exception`.
2. Why does `errorHandler` need to return an `Exception`, if the `RequestState.failure` can receive any object?